### PR TITLE
Split create_new_bounty kwargs building and check if once

### PR DESCRIPTION
**Pull request by mbeacom**
##### Description

The goal of this PR is to reduce the number of checks whether or not `latest_old_bounty` exists and offload the `last_old_bounty` to a class level method for pertinent dict keys/values.

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
bounty creation/updating

##### Testing
Locally via sync_geth

### CODE REVIEW

File: app/dashboard/helpers.py [hunk 1]
```diff
@@ -373,55 +374,54 @@ def create_new_bounty(old_bounties, bounty_payload, bounty_details, bounty_id):
...
+                'web3_created': timezone.make_aware(
                     timezone.datetime.fromtimestamp(bounty_payload.get('created')),
-                    timezone=UTC) if not latest_old_bounty else latest_old_bounty.web3_created,
-                github_url=url if not latest_old_bounty else latest_old_bounty.github_url,
-                token_name=token_name if not latest_old_bounty else latest_old_bounty.token_name,
-                token_address=token_address if not latest_old_bounty else latest_old_bounty.token_address,
-                privacy_preferences=bounty_payload.get('privacy_preferences', {}) if not latest_old_bounty else latest_old_bounty.privacy_preferences,
-                expires_date=timezone.make_aware(
+                    timezone=UTC),
+                'github_url': url,
```
Comment (thelostone-mc) [inline app/dashboard/helpers.py]: 🥇

File: app/dashboard/helpers.py [hunk 2]
```diff
@@ -373,55 +374,54 @@ def create_new_bounty(old_bounties, bounty_payload, bounty_details, bounty_id):
...
-            )
+                    timezone=UTC),
+                'title': bounty_payload.get('title', ''),
+                'issue_description': bounty_payload.get('description', ' '),
+                'balance': bounty_details.get('balance'),
+                'contract_address': bounty_details.get('token'),
+                'network': bounty_details.get('network'),
+                'bounty_type': metadata.get('bountyType', ''),
+                'project_length': metadata.get('projectLength', ''),
+                'experience_level': metadata.get('experienceLevel', ''),
```
Comment (owocki) [inline app/dashboard/helpers.py]: oooh this is much cleaner... yay!

File: app/dashboard/models.py [hunk 3]
```diff
@@ -827,6 +827,41 @@ def additional_funding_summary_sentence(self):
             sentence += f" worth ${afs['usd_value']}"
         return sentence
 
+    def creation_to_dict(self):
+        return {
+            # info to xfr over from self as override fields (this is because sometimes
```
Comment (owocki) [inline app/dashboard/models.py]: is there not a django object to dict() method that we could just use?

Comment (mbeacom) [reply app/dashboard/models.py]: There's a `django.forms` utility method to do this...  I suppose I can use that.

Comment (mbeacom) [reply app/dashboard/models.py]: @owocki Updated to use `django.form.models.model_to_dict`

**Comment by mbeacom** [comment]
@thelostone-mc @owocki Can I have another review of this PR?  I have introduced the `SuperModel.to_standard_dict` method until we update previously implemented `to_dict()` methods on some of the subclasses to maintain parity.  But now, we can simply call `to_standard_dict()` on any of our subclasses of `SuperModel` and get a dictionary representation.  `to_standard_dict` also takes the same two kwargs as `model_to_dict` to allow for customization of the returned dictionary as shown in the one usage example so far.

File: app/dashboard/helpers.py [hunk 4]
```diff
@@ -373,54 +374,64 @@ def create_new_bounty(old_bounties, bounty_payload, bounty_details, bounty_id):
...
+                'permission_type': bounty_payload.get('schemes', {}).get('permission_type', 'permissionless'),
+                'attached_job_description': bounty_payload.get('hiring', {}).get('jobDescription', None),
+                'bounty_owner_github_username': bounty_issuer.get('githubUsername', ''),
+                'bounty_owner_address': bounty_issuer.get('address', ''),
+                'bounty_owner_email': bounty_issuer.get('email', ''),
+                'bounty_owner_name': bounty_issuer.get('name', ''),
+            })
+        else:
+            latest_old_bounty_dict = latest_old_bounty.to_standard_dict(
+                fields=[
```
Comment (owocki) [inline app/dashboard/helpers.py]: is `fields` used here?

Comment (owocki) [reply app/dashboard/helpers.py]: oh wait nvm i see it now

**Comment by mbeacom** [comment]
https://stage.gitcoin.co/issue/mbeacom/test-repo/64/979

**Comment by owocki** [comment]
:shipit:

**Pull request merged**